Client/core/agent.py

```
@@ -0,0 +1,108 @@
+        except Exception as e:
+            logging.error(f"Agent encountered a fatal error: {e}")
+            traceback.print_exc()
+        finally:
+            self.stop()
+
+    def stop(self):
+        logging.info("Stopping Agent...")
+        self.running = False
+        self.model_loader.unload_models()
+
+    def _main_loop(self):
```

Ruff (0.15.0)

[warning] Missing return type annotation for private function `_main_loop`

Add return type annotation: `None`

(ANN202)